src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -353,6 +356,44 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 		this._register(disposable);
 	}
 
+	async showProfileMigrationNotification(): Promise<void> {
+		const platform = this._getPlatformKey();
+		const shouldMigrateToProfile = (!!this._configurationService.getValue(TerminalSettingPrefix.Shell + platform) ||
+			!!this._configurationService.getValue(TerminalSettingPrefix.ShellArgs + platform)) &&
+			!!this._configurationService.getValue(TerminalSettingPrefix.DefaultProfile + platform);
+		if (shouldMigrateToProfile && this._storageService.getBoolean(SHOULD_PROMPT_FOR_PROFILE_MIGRATION_KEY, StorageScope.WORKSPACE, true)) {
+			this._notificationService.prompt(
+				Severity.Info,
+				nls.localize('terminalProfileMigration', "You are using deprecated shell/shellArgs settings, but have a defaultProfile set. Want to create a profile from those and use that instead?"),
+				[
+					{
+						label: nls.localize('switchToProfile', "Yes"),
+						run: async () => {
+							const shell = this._configurationService.getValue(`${TerminalSettingPrefix.Shell + platform}`);
+							const shellArgs = this._configurationService.getValue(`${TerminalSettingPrefix.ShellArgs + platform}`);
+							const profile = await this._terminalProfileResolverService.createProfileFromShellAndShellArgs(shell, shellArgs);
+							if (profile) {
+								this._configurationService.updateValue(`${TerminalSettingPrefix.DefaultProfile + platform}`, profile.profileName);
+								this._configurationService.updateValue(`${TerminalSettingPrefix.Shell + platform}`, null);
+								this._configurationService.updateValue(`${TerminalSettingPrefix.ShellArgs + platform}`, null);
+								this._logService.trace(`migrated from shell/shellArgs, ${shell} ${shellArgs} to profile ${JSON.stringify(profile)}`);
```

Comment:
shellArgs probably won't get logged right

Reply:
they're just a string, array, or null. will that not work? when I tested, it seemed  fine